### PR TITLE
Add Keychron Q8

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
Add new keyboard q8 which contains four sub versions
Thanks

## Description

- Add ANSI and ISO versions, including encoder variants

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by kazigk** [comment]
Readme for that keyboard states the following:
`- Turn clockwise to increase volume and turn anti-clockwise to decrease volume.`

Rotary encoder seems to be reversed. It's clockwise for volume down and counterclockwise for volume up, it should be the opposite.

I tested on `ansi_encoder` variant with `keychron` keymap.

**Comment by lalalademaxiya1** [comment]
> 

Thanks, i also found that the pin define of encoder is reverse and i fixed it.

**Review by fauxpark — APPROVED** [review]

File: keyboards/keychron/q8/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+A customizable 65% Alice layout keyboard.
+
+* Keyboard Maintainer: [Keychron](https://github.com/keychron)
+* Hardware Supported: Keychron Q8
+* Hardware Availability: [Keychron](https://www.keychron.com)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make keychron/q8/q8_ansi_stm32l432:default
```
Comment (coliss86) [inline keyboards/keychron/q8/readme.md:11]: Is this name correct ? I don't see any folder `keychron/q8/q8_ansi_stm32l432` but only `keychron/q8/q8_ansi`?

Comment (adophoxia) [reply keyboards/keychron/q8/readme.md:11]: Should just be `ansi`.

Comment (lalalademaxiya1) [reply keyboards/keychron/q8/readme.md:11]: Thanks, corrected now.

**Pull request merged**